plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -193,18 +193,34 @@ public function update( &$order ) {
 		// Update the order.
 		parent::update( $order );
 
-		// Fire a hook depending on the status - this should be considered a creation if it was previously draft status.
 		$new_status = $order->get_status( 'edit' );
 
-		if ( $new_status !== $previous_status && in_array( $previous_status, array( 'new', 'auto-draft', 'draft', 'checkout-draft' ), true ) ) {
+		// We need to remove the wc- prefix from the status for comparison and proper evaluation of new vs updated orders.
+		if ( strpos( $previous_status, 'wc-' ) === 0 ) {
+			$previous_status = substr( $previous_status, 3 );
+		}
+
+		if ( strpos( $new_status, 'wc-' ) === 0 ) {
+			$new_status = substr( $new_status, 3 );
+		}
+
+		$nonTriggeringOrderStatuses = array( 'new', 'auto-draft', 'draft', 'checkout-draft', 'refunded', 'failed', 'cancelled');
```

Comment:
Check your IDE is setup to show linting rules. This violates naming (snake_case is needed on the PHP side).

Reply:
It is highlighting, I don't know what glitch got into it or my brain 🫠
Changed in https://github.com/woocommerce/woocommerce/pull/49098/commits/3f28c6b664885f7ee475803df85b1766eca56dc7